.github/codecov.yml

```diff
@@ -0,0 +1,7 @@
+coverage:
+  status:
+    project:
+      default:
+        target: auto
+        threshold: 1%
+        if_ci_failed: success
```

Comment:
Does `if_ci_failed: success` mean that the PR will be marked successful even if the CI Failed?
Also, by `success` will the PR be marked green? Or is the `success` status exclusive to `Codecov` ?

Reply:
Also, do you think we could configure codecov for [removed_code_behavior](https://docs.codecov.com/docs/commit-status#removed_code_behavior) ? Just bouncing ideas.

Reply:
Yes it means that the PR will be marked successful but only if codecov fails. This is a part of `codecov.yml` so it shouldn't affect other CI jobs.

Reply:
> Also, do you think we could configure codecov for [removed_code_behavior](https://docs.codecov.com/docs/commit-status#removed_code_behavior) ? Just bouncing ideas.

Ohh yeah that's a good point! I think the default option of `fully_covered_patch` would be good in our case as it only considers the patch, and I don't think we should factor in removed code as increased test coverage.

Reply:
> Yes it means that the PR will be marked successful but only if codecov fails.

I am not sure if I a getting this right, what are the scenarios where the code coverage test could fail? 
I am wondering if codecov's CI implies one of the CI providers as listed in here https://docs.codecov.com/docs/supported-ci-providers ?

Reply:
Also, I am not sure what category does our CI fall into? Is it `prow` or `GitHub Actions`? cc: @CecileRobertMichon @nojnhuh

Reply:
In this case, we're using GitHub actions as the CI provider. The code coverage test fails if the coverage diff is greater than the threshold of 1% like so: https://github.com/willie-yao/cluster-api-provider-azure/pull/2

Right now, I'm working on a fix to not mark the action as failed, but still display the message that the threshold has been exceeded.